src/bin/commands/demux.rs

```diff
@@ -213,7 +214,12 @@ impl ReadSet {
             None => {
                 // If no pre-existing comment, assume the read is a passing filter, non-control
                 // read and generate a comment for it (sample barcode is added below).
-                write!(writer, "{}:N:0:", read_num)?;
+                if read_num < Self::READ_NUMBERS.len() {
```

Comment:
1. Same comment as below about `<` -> `<=` if you don't start at 0.
2. Probably worth leaving a comment about performance and why we want an optimization here.